Process issue action does not generate files in project/data

Since the source YAML file is being modified, the `make all-data` command should be run afterwards to ensure that the JSON/TSV files are generated with the updated data.

@caufieldjh Does `make site` work for you? When running, I am getting some errors. For example:

```
sed: -e: No such file or directory
Copying individual topic metadata to each topic page...
mkdocs build ;
bash: line 1: mkdocs: command not found
make: *** [site] Error 127
```

--- a/.github/scripts/process_issue.py
+++ b/.github/scripts/process_issue.py
@@ -46,6 +46,7 @@ import itertools as itt
 import logging
 import os
 import re
+import subprocess
 import sys
 import time
 from subprocess import CalledProcessError, check_output
@@ -499,6 +500,13 @@ def main(dry: bool, github: bool, force: bool):
 
     click.secho("Creating and switching to branch", fg="green")
     click.echo(branch(branch_name))
+
+    try:
+        click.secho("Running `make site`", fg="green")
+        subprocess.run(["make", "site"], check=True)
+    except subprocess.CalledProcessError as e:
+        click.secho(f"An error occurred while running `make site`: {e}", fg="red")
+        return sys.exit(0)
 
     click.secho("Committing", fg="green")
     commit_msg = commit_all(message)

